llmfoundry/models/layers/blocks.py

```diff
@@ -212,13 +212,32 @@ def apply_ffn(
         indices = None
         if not self.use_pad_tok_in_ffn and attention_mask is not None:
             assert unpad_input is not None
+            attention_mask = self.slice_attention_mask(attention_mask, seq_len)
```

Comment:
I'm confused, why would we use the `slice_attention_mask` function in `apply_ffn` for the MPTBlock? What's an example use case?

Reply:
one might want to slice the attention mask differently for different parallelism techniques.